frontend/src/components/common/FractionWrapper.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Divider, Input, VStack } from "@chakra-ui/react";
+
+interface FractionWrapperProps {
+  denominator: string;
+  numerator: string;
+  onNumeratorChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onDenominatorChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  readOnly?: boolean;
+}
+
+const getFractionInputWidth = (value: string): number => {
+  return value.length > 1 ? 48 + 8 * value.length : 48;
+};
+
+const getDividerWidth = (width1: number, width2: number) => {
+  return Math.max(width1, width2) + 12;
+};
+
+const FractionWrapper = ({
+  denominator,
+  numerator,
+  onNumeratorChange,
+  onDenominatorChange,
+  readOnly = false,
+}: FractionWrapperProps): React.ReactElement => {
+  const numeratorWidth = getFractionInputWidth(numerator);
+  const denominatorWidth = getFractionInputWidth(denominator);
+
+  const FractionInput = ({
```

Comment:
nit: I'd prefer to define components at the top level, since the nesting can make it a bit hard to reason about things.